### DISCUSSION TITLE
issue w/ generate_derivatives task

### DISCUSSION THREAD

**Discussion by hermanem** [CollectionBuilder/collectionbuilder.github.io#127] [General]
I've been trying to run the 'rake generate_derivatives' task in VS Code, but the task keeps aborting with the following error message: 
<img width="555" alt="Screenshot 2024-06-05 at 11 06 28 AM" src="https://github.com/CollectionBuilder/collectionbuilder.github.io/assets/165897965/c8fa4977-562b-4845-a5ef-9d342e37b52b">
I've tried running the task with my own objects and with the demo objects, and I get the same result either way. I've downloaded all of the necessary software, and I've checked and rechecked the rakefile to make sure I haven't inadvertently altered anything. Any help would be much appreciated.

**Comment by evanwill** [comment]
@hermanem are you on Windows or Mac?
Did you do a `bundle install` in your project repository? 
That should install `image_optim` package if you are on Mac or Linux. It skips it on Windows because it isn't supported (that is what the Rake task doing with the `unless Gem.win_platform?`, skipping over optimization on Windows).

**Reply by hermanem** [reply]
I'm on a Mac (Sonoma 14.5). I ran 'bundle install' in VS Code, and I seem to have made some progress. However, now I get this error message for each of my objects: 
<img width="693" alt="Screenshot 2024-06-05 at 2 21 46 PM" src="https://github.com/CollectionBuilder/collectionbuilder.github.io/assets/165897965/5760742a-d3b5-4e92-853b-11ddd3e2b36c">
I've installed, uninstalled, reinstalled imagemagick and ghostscript. Is there a command I need to run in VS Code like I did with your previous instruction? Thanks for your help so far.

**Reply by evanwill** [reply]
It looks like mini_magick is not finding ImageMagick on your path. It might be that your terminal in VS Code needs to be restarted. 
First, open a terminal window outside of VS Code, by searching in Spotlight for terminal. Then type 
`magick --version`
If installed correctly, you should get something back about which version you have installed. 
If you get an error, try restarting your laptop to see if it just needs to refresh the path. If you still get an error, it means something is up with your magick install. 

However, if `magick --version` in your system terminal looks correct, then the issue is that VS Code's terminal isn't refreshing. 
First, in VS Code, in Settings, search for “persistent”, then uncheck the option “Terminal > Integrated: Enable Persistent Sessions”--this seems to cause mysterious problems with updating things in the terminal. Then, on VS Code, on the terminal window, click the "trash" icon to fully close down the terminal, then fully quit restart VS Code.

**Reply by hermanem** [reply]
The second thing you mentioned was the issue. Everything seems to be good now. Thanks for your help!
